### ISSUE TITLE
Add support for Google disk snapshots

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#438]
Terraform resource: [google_compute_snapshot](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_snapshot), the resource is already in the codebase, just returns nil as it's not been implemented.

We can add one cost component "Name: Storage" with "Units: GB-months". The following Pricing API filters should be used. The `storage_gb` infracost-usage.yml file parameter should be added/used too. The `google_compute_disk_resource_policy_attachment` resource looks like it's free, we can add it to the free resource list as part of this ticket too.

```
    "productFilter": {
      "service": "Compute Engine",
      "productFamily": "Storage",
      "region": "us-east1",
      "attributeFilters": [
				{
          "key": "description",
          "value": "Storage PD Snapshot"
        }
      ]
    }
  }
```

Pricing details: see [here](https://cloud.google.com/compute/disks-image-pricing?skip_cache=true#disk), "Snapshot storage" is the relevant price. Cloud Storage already includes the network transfer prices so we don't need to include those in the snapshot resource.

I can't find the pricing filters for "Multi-regional snapshot storage" using the following, @aliscott any ideas where it might be hiding?
```
db.products.distinct("attributes.description", {"vendorName": "gcp", "region": "us-east1", "service": "Compute Engine"})

db.products.distinct("attributes.description", {"vendorName": "gcp", "region": "us-east1", "service": "Compute Engine", "productFamily": "Storage"})

db.products.distinct("attributes.description", {"vendorName": "gcp", "region": "us-east1", "service": "Compute Engine", "productFamily":"Compute"})
```

**Comment by ekifel** [comment]
@alikhajeh1 hey, I want to work on it!

**Comment by alikhajeh1** [comment]
Released in v0.8.3
